chartfx-chart/src/main/java/de/gsi/chart/plugins/TableViewer.java

```diff
@@ -227,28 +234,64 @@ protected HBox getInteractorBar() {
         private Callable<Void> refreshFunction;
 
         private final ListChangeListener<Renderer> rendererChangeListener = this::rendererChanged;
-        private final ListChangeListener<DataSet> datasetChangeListener = this::datasetsChanged;
-        private final EventListener dataSetDataUpdateListener = (UpdateEvent evt) -> FXUtils.runFX(() -> {
-            nRows = 0;
-            for (TableColumn<DataSetsRow, ?> col : columns) {
-                if (col instanceof DataSetTableColumns) {
-                    DataSetTableColumns dataSetColumn = ((DataSetTableColumns) col);
-                    nRows = Math.max(nRows, dataSetColumn.dataSet.getDataCount(DIM_X));
-                    for (final TableColumn<DataSetsRow, ?> subColumn : dataSetColumn.getColumns()) {
-                        if (subColumn instanceof DataSetTableColumn) {
-                            ((DataSetTableColumn) subColumn).updateEditableState();
-                        }
-                    }
-                }
-            }
-            refresh();
-        });
+        private final InvalidationListener datasetChangeListener = this::datasetsChanged;
+        private final EventListener dataSetDataUpdateListener = (UpdateEvent evt) -> FXUtils
+                .runFX(() -> this.datasetsChanged(null));
 
         public DataSetsModel() {
             super();
             columns.add(new RowIndexHeaderTableColumn());
         }
 
+        @Override
+        public String toString() {
+            return "TableModel";
+        }
+
+        private long lastColumnUpdate = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4834981)